packages/components/src/components/dialog/dialog.component.ts

```diff
@@ -0,0 +1,317 @@
+   */
+  @property({ type: String, reflect: true, attribute: 'header-text' })
+  headerText: string = '';
+
+  /**
+   * Defines a string value to display as the under-header description of the dialog
+   */
+  @property({ type: String, reflect: true, attribute: 'description-text' })
+  descriptionText: string = '';
+
+  /**
+   * The html tag to be used for the header text
+   */
+  @property({ type: String, reflect: true, attribute: 'header-tag-name' })
+  headerTagName: string = DEFAULTS.HEADER_TAG_NAME;
+
+  /**
+   * The html tag to be used for the below-header description text
+   */
+  @property({ type: String, reflect: true, attribute: 'description-tag-name' })
+  descriptionTagName: string = DEFAULTS.DESCRIPTION_TAG_NAME;
+
+  /**
+   * Role of the dialog
+   * @default dialog
+   */
+  @property({ type: String, reflect: true })
+  override role: DialogRole = DEFAULTS.ROLE;
+
+  public triggerElement: HTMLElement | null = null;
+
+  /** @internal */
+  private utils: DialogUtils;
+
+  /** @internal */
+  public backdropElement: HTMLElement | null = null;
+
+  constructor() {
+    super();
+    this.utils = new DialogUtils(this);
+  }
+
+  protected override async firstUpdated(changedProperties: PropertyValues) {
+    super.firstUpdated(changedProperties);
+    this.setupTriggerListener();
+    this.utils.setupAccessibility();
+    this.style.zIndex = `${this.zIndex}`;
+    DialogEventManager.onCreatedDialog(this);
+
+    if (this.visible) {
+      await this.handleCreateDialogFirstUpdate();
+    }
+  }
+
+  override async disconnectedCallback() {
+    super.disconnectedCallback();
+    this.removeEventListeners();
+    DialogEventManager.onDestroyedDialog(this);
+  }
+
+  /**
+   * Sets up the trigger listener for focus trap
+   */
+  private setupTriggerListener() {
+    if (!this.triggerId) return;
+
+    this.triggerElement = (this.getRootNode() as Document | ShadowRoot).querySelector(`[id="${this.triggerId}"]`);
+    if (!this.triggerElement) return;
+
+    this.addEventListener('focus-trap-exit', this.hideDialog);
+  }
+
+  /**
+   * Removes the trigger event listener
+   */
+  private removeEventListeners() {
+    if (!this.triggerElement) return;
+
+    this.removeEventListener('focus-trap-exit', this.hideDialog);
+  }
+
+  protected override async updated(changedProperties: PropertyValues) {
+    super.updated(changedProperties);
+
+    if (changedProperties.has('visible')) {
+      const oldValue = (changedProperties.get('visible') as boolean | undefined) || false;
+      await this.isOpenUpdated(oldValue, this.visible);
+    }
+    if (changedProperties.has('zIndex')) {
+      this.setAttribute('z-index', `${this.zIndex}`);
+    }
+    if (
+      changedProperties.has('ariaLabel')
+      || changedProperties.has('ariaLabelledById')
+    ) {
+      this.utils.setupAccessibility();
+    }
+  }
+
+  /**
+   * Handles the escape keydown event to close the dialog.
+   *
```

Comment:
```suggestion
   * @internal
```

This would hide the function on storybook.